test/org/parosproxy/paros/core/scanner/VariantJSONQueryTest.java

```diff
@@ -0,0 +1,75 @@
+package org.parosproxy.paros.core.scanner;
+
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit test for {@link VariantJSONQuery}.
+ */
+public class VariantJSONQueryTest {
+    @Test
+    public void shouldHaveParametersListEmptyByDefault() {
+        // Given
+        VariantJSONQuery variantJSONQuery = new VariantJSONQuery();
+        // When
+        List<NameValuePair> parameters = variantJSONQuery.getParamList();
+        // Then
+        assertThat(parameters, is(empty()));
+    }
+
+    @Test
+    public void shouldFindStringValueInContent(){
+        // Given
+        VariantJSONQuery variantJSONQuery = new VariantJSONQuery();
+        variantJSONQuery.setMessage(getMessageWithBody("{\"foo\":\"bar\"}"));
+        // When
+        List<NameValuePair> parameters = variantJSONQuery.getParamList();
+        // Then
+        assertThat(parameters.get(0).getName(), is("foo"));
+        assertThat(parameters.get(0).getValue(), is("bar"));
+    }
+
+    @Test
+    public void shouldFindStringWithEscapedCharacter(){
+        // Given
+        VariantJSONQuery variantJSONQuery = new VariantJSONQuery();
+        variantJSONQuery.setMessage(getMessageWithBody("{\"foo\":\"bar\n\"}"));
+        // When
+        List<NameValuePair> parameters = variantJSONQuery.getParamList();
+        // Then
+        assertThat(parameters.get(0).getName(), is("foo"));
+        assertThat(parameters.get(0).getValue(), is("bar\n"));
+    }
+
+    @Test
+    public void shouldFindStringWithEscapedQuotationMark(){
+        // Given
+        VariantJSONQuery variantJSONQuery = new VariantJSONQuery();
+        variantJSONQuery.setMessage(getMessageWithBody("{\"foo\":\"bar\\\"\"}"));
+        // When
+        List<NameValuePair> parameters = variantJSONQuery.getParamList();
+        // Then
+        assertThat(parameters.get(0).getName(), is("foo"));
+        assertThat(parameters.get(0).getValue(), is("bar\""));
+    }
+
+    private HttpMessage getMessageWithBody(String body){
+        try {
+            return new HttpMessage(
+                    new HttpRequestHeader("POST / HTTP/1.1\nHost: www.example.com\nContent-Type: application/json"),
+                    new HttpRequestBody(body));
+        } catch(HttpMalformedHeaderException e) {
```

Comment:
It's preferable to re-thrown the exception (or don't catch at all). This might start to fail if the parsing is changed (e.g. be stricter).